src/main/java/allocator/Allocator.java

```diff
@@ -5,20 +5,45 @@
 import university.University;
 import university.UniversityRepository;
 
+import java.util.Set;
+
+import exceptions.SEPException;
+
+import java.util.HashSet;
+
 public class Allocator {
     
+    private static double minimumGPA = 0;
     private StudentList studentList;
+    
 
     public Allocator(StudentList studentList) {
         this.studentList = new StudentList(studentList);  // don't want to modify the original list
     }
+
+    public void setMinimumGPA(String stringGpa) throws SEPException{
+        Set<String> errorMessages = new HashSet<>();
+
+        float gpa = studentList.validateGpa(stringGpa, errorMessages);
+
+        if (!errorMessages.isEmpty()) {
+            throw new SEPException(String.join("\n", errorMessages));
+        }
+
+        //Setting the static minimum GPA criteria to the input
+        minimumGPA = gpa;
+    }
+
+    public double getMinimumGPA(){
+        return minimumGPA;
+    }
     
     public StudentList allocate() {
         studentList.sortStudentsByGPA();
         for (Student student : studentList.getList()) {
             for (int uni : student.getUniPreferences()) {
                 University university = UniversityRepository.getUniversityByIndex(uni);
-                if (university.getSpotsLeft() > 0) {
+                if (university.getSpotsLeft() > 0 && student.getGpa() >= minimumGPA) {
```

Comment:
My idea of minimumGpa was for it to be a member of each `University` object, but at the moment setting the criteria as a `Allocator` level variable seems to be fine as well. 👍

Reply:
Thanks!